### DISCUSSION TITLE
Adding ExoPlayer to Uno WinUI porject

### DISCUSSION THREAD

**Discussion by LiphiTC** [unoplatform/uno#11140] [Q&A] [answered]
I've created a new Uno WinUI .net 7 project and imported a [Xamarin binding for Exoplayer](https://github.com/Baseflow/ExoPlayerXamarin) to the Mobile project
```
<Page
    x:Class="Syncplay.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:android="http://uno.ui/android"
    mc:Ignorable="d android"   
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
    
    
    <Grid>
      <android:com.google.android.exoplayer2.ui.StyledPlayerView android:id="@+id/player_view_sample"/>
    </Grid>
    
</Page>
```
Rider didn't like that I added android:id attribute for element
![image](https://user-images.githubusercontent.com/48130941/216051377-9e052470-cd3d-43af-ae1a-56f72a97be59.png)

But I ignored it and proceeded to build.... and got a memory leak in dotnet 
![image](https://user-images.githubusercontent.com/48130941/216051988-e23d9210-64a9-42e9-b089-7227b38ae454.png)

Memory usage just keeps going until someone or something doesn't kill it.

So is there any way to get an element from code?

**Comment by jeromelaban** [comment, marked as answer]
`<android:com.google.android.exoplayer2.ui.StyledPlayerView android:id="@+id/player_view_sample"/>` is not a valid XAML for Uno Platform.

You can try using `<android:StyledPlayerView />` instead.

You can integrate native controls [this way](https://platform.uno/docs/articles/platform-specific-xaml.html).

**Reply by LiphiTC** [reply]
Oh okay. Thank you.
